DMX Output support

hdanak opened the pull request:
Still need to clean up a bunch of things. I ended up switching to using separate model classes rather than the builder. Also using the new LX submodel features which let you filter by "keys", which are basically tags.

hdanak committed:
Significant model refactoring and quick and dirty DMX output.

hdanak committed:
Hacks to make output possibly work.

hdanak committed:
Add more flexible DMX output code.

hdanak commented:
@somaholiday I've refactored the DMX output code and added constants `SPOTLIGHT_DMX_START_CHANNEL` and `WALL_WASHER_DMX_START_CHANNEL` which lets you start the wall washers at a different channel than where the spotlights end.

hdanak committed:
Remove unused code from old model builder.

somaholiday commented on src/main/java/org/yomigae/model/TempleModel.java:
```diff
@@ -30,142 +30,186 @@
+		}
 
-		private static ToriiType[] values = values();
+		submodels.add(createTorii(direction, ToriiModel.ToriiType.T2, t, hallKeys));
```
What's the motivation for T2 to be pulled out by itself, rather than being included in the loop above?

hdanak replied:
Oh I think it was because in the builder it was being handled differently, but no longer needs to be.

somaholiday commented on src/main/java/org/yomigae/model/TempleModel.java:
```diff
@@ -30,142 +30,186 @@
+		return filteredPoints;
+	}
+
+	public Set<LXPoint> filterClusterPoints(Set<FilterFlags> flags) {
```
When do you think we'll need to be able to filter out clusters, rather than individual points?

somaholiday commented on src/main/java/org/yomigae/model/SpotlightModel.java:
```diff
@@ -0,0 +1,65 @@
+import heronarts.lx.model.LXPoint;
+import heronarts.lx.transform.LXTransform;
+
+public class SpotlightModel extends LXModel {
```
This is totally pedantic, but since I've been reading a lot about this lately: technically what we're using are PAR washers--a spotlight is actually a different kind of light.

I have no strong compulsion to change the class name based on this. :P

somaholiday committed:
Update SpotlightDmxFragment

Also increment buffer offset after applying each DmxFragment in DmxAggregate

hdanak committed:
Remove special case for T2 torii in temple construction.

somaholiday committed:
Update WallWasherDmxFragment

Both SpotlightDmxFragment and WallWasherDmxFragment now take into account their starting channels.

The resulting sACN datagrams are well formed and have been verified in sACNView.

The physical fixtures themselves are displaying colors correctly, but are still flickering for some reason.
